### DISCUSSION TITLE
testing a composable that contains $fetch or useFetch

### DISCUSSION THREAD

**Discussion by keivanzang** [nuxt/nuxt#15920] [Questions]
I have a composable like this 

`export function get(uri) {
    return $fetch(uri);
}`

how do you handle unit tests like this in runtime without mocking and going for browser testing (its slow)
also the server is not nuxt(nitro) it uses dedicated backend
the best result was able to achieve was some error like 'nuxt instance not found'
i tried almost all testing methods across web that i could get my hand on but it seems like
its just not testable this way

**Comment by TheAlexLichter** [comment]
What speaks against mocking or mocking the request via [MSW](https://mswjs.io/) (as long as it is not rendered through the server)?

**Reply by keivanzang** [reply]
I just prefer to not use it in this particular case because of refactoring purposes. 

Also i don't quite have clear vision on SSR tests and how they are done if thats the only way to go about it then fine.

Im not sure even mocking can help here at all because i don't see anything here to mock (just saying).

In this simplified version of the problem all it need is to make a call to a server and get the result and these all need to happens in server.
